src/easylink/rule.py

```diff
@@ -125,14 +126,26 @@ def build_rule(self) -> str:
     def _build_io(self) -> str:
         """Builds the input/output portion of the rule."""
         log_path_chunk_adder = "-{chunk}" if self.is_embarrassingly_parallel else ""
+        # Handle output files vs directories
+        files = [path for path in self.output if Path(path).suffix != ""]
```

Comment:
This is the main focus of this PR - it allows for directories to be the output of the snake step rather than files.

NOTE: this makes the assumption that if pathlib.Path().suffix is empty, then it is a directory. This means that if a step is saving out a file w/o an extension, things will break. I didn't know what else to do, though, since these are paths that do not yet exist, so we can't, like, .isdir().

Reply:
I like this better than the alternatives I had imagined